snmp/tests/test_profiles.py

```diff
@@ -48,6 +49,17 @@
 pytestmark = pytest.mark.usefixtures("dd_environment")
 
 
+def test_load_profiles(aggregator, caplog):
+
+    instance = common.generate_instance_config([])
+    check = SnmpCheck('snmp', {}, [instance])
+    for name, profile in check.profiles.items():
+        try:
+            check._config.refresh_with_profile(profile)
+        except ConfigurationError as e:
+            pytest.fail("Profile `{}` is not configured correctly: {}".format(name, e))
```

Comment:
I'm not sure if printing the exception message here is necessary since it's printed earlier in the stack trace

Reply:
The traceback doesn't look useful to me — it's generic and not particular to a profile, so it won't necessarily help with debugging for people getting it.

So maybe we can keep the error message here and add `__tracebackhide__`?  Then it's still possible to pass `-pa --full-trace` if we're interested in the traceback.

https://docs.pytest.org/en/latest/example/simple.html#writing-well-integrated-assertion-helpers

```python
def _verify_profile(check, profile):
    __tracebackhide__ = True
    try:
        check._config.refresh_with_profile(profile)
    except ConfigurationError as e:
        pytest.fail("Profile `{}` is not configured correctly: {}".format(name, e))

def test_load_profiles(aggregator):
    instance = common.generate_instance_config([])
    check = SnmpCheck('snmp', {}, [instance])
    for name, profile in check.profiles.items():
        _verify_profile(check, profile)
```

Reply:
Sure, that sounds good!

Reply:
Updated to remove the duplicated messages!